src/operator/optimizer_op-inl.h

```diff
@@ -1639,8 +1639,8 @@ struct LambUpdatePhaseOneKernel {
     DType g = mean_data[i] / (square_root::Map(var_data[i]) + epsilon) + wd * weight_data[i];
 
     if (bias_correction) {
-      DType mean_hat = mean_data[i] / (1. - power::Map(beta1, t));
-      DType var_hat = var_data[i] / (1 - power::Map(beta2, t));
+      DType mean_hat = mean_data[i] / (1. - std::pow(beta1, t));
```

Comment:
Can we still re-use the power::Map() data structure here with a new signature for Map()?

Reply:
@apeforest  i'm not sure mshadow allows mixed data type as inputs